### DISCUSSION TITLE
Request: Add this Dreambooth extension for auto1111

### DISCUSSION THREAD

**Discussion by dathide** [AbdBarho/stable-diffusion-webui-docker#208] [Ideas]
Thanks again for your work keeping this repo up to date.

This new extension went public yesterday: https://github.com/d8ahazard/sd_dreambooth_extension

d8ahazard has worked hard getting Dreambooth working with AUTOMATIC1111's UI and mid-range GPUs. Instructions are provided on how to install it. I think it should be enabled by default (after your testing of course) since Dreambooth is so powerful and useful. d8ahazard has already been hard at work fixing any bugs.

**Comment by DevilaN** [comment]
Why should it be enabled by default when A1111 repo is not incorporating it right now other than using it as an extension?
Is there any difficulty with it?

**Reply by dathide** [reply]
Auto gives his reasons in this comment: https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/3995#issuecomment-1298741868

Basically it's just that it's a lot of code that Auto would need to study to incorporate it. It's not that there are issues with it.

**Comment by AbdBarho** [comment]
Extensions are a tough sell, if I add an extension, I have to basically do IT support for it, which is like auto said, code I didn't write and don't want to maintain.

The UI provides an extremely easy way of installing extensions, and `startup.sh` can be used to do any magic you deem necessary.

**Reply by dathide** [reply]
I thought it would be an easy addition, but it's understandable to not include it if it adds significant extra work to maintain. Anyways, I'm glad that it's easy for me to install on my own.

**Comment by dathide** [comment]
To get it working, I added `--enable-insecure-extension-access` to CLI_ARGS for AUTOMATIC1111 in docker-compose.yml. The extension then successfully installs, but when I restarted the UI, an error about accelerate not being installed showed up. So I went into AUTO's Dockerfile and added `pip install accelerate` right after the last `pip install -r requirements_versions.txt`. The extension then loads successfully, but accelerate isn't configured. Is there a way for me to run `accelerate config`? In [this docker repo](https://github.com/Sharrnah/dreambooth-docker), there is an accelerate-config.yaml and the command to configure accelerate is `docker compose -f docker-compose.build.yaml exec dreambooth accelerate config --config_file=/app/examples/dreambooth/accelerate-config.yaml`. Would this be a difficult thing to add to this repo and include accelerate in the default installation?

**Reply by AbdBarho** [reply]
@dathide I hope you can see now why I don't want to add any extensions to this repo.

you can create a script file `data/config/auto/startup.sh` (which will run on container startup) and put whatever you want in it, install accelerate, configure it, copy files, etc...

**Reply by gowgzies** [reply]
@dathide Let me know if you figure this out, and thanks for your efforts!

**Reply by olivervhansen** [reply]
@dathide did you have any luck with Dreambooth? I got it working but had to make some changes to the running container. I'm not exactly sure the best way to make the changes stick when the container is rebuilt.
My `--enable-insecure-extension-access` was already there in docker-compose.yml but in the container I ran:
` pip install -U diffusers` which upgraded diffusers from 0.3.0 to 0.9.0
`pip install --upgrade transformers==4.22.1` to upgrade transformers from 4.19.2 to 4.22.1

I think that's all but there could have been more. Is it just including those lines in the data/config/auto/startup.sh script? Or should those packages be updated through some of the existing requirements.txt files?
